ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -716,41 +716,36 @@ private void throwIfUnsupported(final Query query) {
     }
     if (query.getWhere().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support WHERE."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (query.getGroupBy().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support GROUP BY."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (query.getHaving().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support HAVING."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (query.getWindow().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support WINDOWS."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (query.getPartitionBy().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support PARTITION BY."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (query.getLimit().isPresent()) {
       throw new UnsupportedOperationException("New query planner does not support LIMIT."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     final Relation fromClause = query.getFrom();
     if (fromClause instanceof Join) {
       throw new UnsupportedOperationException("New query planner does not support joins."
-          + "Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
+          + " Set " + KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED + "=false.");
     }
     if (fromClause instanceof JoinedSource) {
       throw new IllegalStateException(); // top level node should always be Join
     }
-    if (query.getSelect().getSelectItems().size() > 2
```

Comment:
Since this PR is for just the "simple column", does it make sense to continue checking any restrictions here?

Reply:
I think we cannot do a good check here any longer, without writing a log of code. Better to push checks down into the "select node" class.